src/ort_genai_c.h

```diff
@@ -278,6 +278,7 @@ OGA_EXPORT void OGA_API_CALL OgaDestroyGenerator(OgaGenerator* generator);
  * \return True if the generator has finished generating all the sequences, false otherwise.
  */
 OGA_EXPORT bool OGA_API_CALL OgaGenerator_IsDone(const OgaGenerator* generator);
+OGA_EXPORT bool OGA_API_CALL OgaGenerator_IsSessionTerminated(const OgaGenerator* generator);
```

Comment:
I agree that it is sort of defeating the purpose of key value pair, but I believe this is unique to set terminate feature because the user will need to know the difference between a normal error and an error generated because of session being in terminated state. 

If you believe that the GetRunOption would be usable for other features as well, I can modify this, so please let me know.

Reply:
Also, if there is a better way for user to identify that the session is terminated that would also work.

Reply:
Thanks Baiju for bringing this again, if I remember correctly from our last conversation on this, Error codes might take a bit longer for this work, am I correct?

Reply:
Ok, let's keep this. We'll fix this when we revise the APIs.

Reply:
Sure, let me create an item for this.